### DISCUSSION TITLE
Next/image blurPlaceholder skipped on virtualizer update

### DISCUSSION THREAD

**Discussion by michael-mcgreal** [TanStack/virtual#412] [Q&A]
Using nextJS Image component with `placeholder='blur'` and a blurDataUrl of base64.

_Both examples have cache disabled and a throttled network for exaggeration_
<details>
<summary>Expected behavior: (mapping over data with JSX)</summary>

Blur placeholder base64 image shown on component mount *and* on list update (when dragging the map)

https://user-images.githubusercontent.com/11381911/197398592-248ef354-f873-494c-b420-14d7bf041bfc.mp4

![mapping data to jsx (expected blur behavior)](https://user-images.githubusercontent.com/11381911/197398620-f88eea3d-449e-4967-9490-4e2f9289671b.png)

</details>

<details>
<summary>Problem behavior: Virtualization skips blurPlaceholder, somehow showing old image until new image loads. </summary>

The blurPlaceholder is shown on react-virtual mount, but on list update: the old image is shown until the new image has loaded (skipping blurPlaceholder).  

The blurPlaceholder base64 data is included in the same json as the text – available immediately (it's not a remote url).

https://user-images.githubusercontent.com/11381911/197399901-893ea74e-902f-43ac-9353-e34a9c66bbaa.mp4

![react-virtual](https://user-images.githubusercontent.com/11381911/197399914-f30f0eb6-405e-4810-92de-a7c581394f47.png)

</details>

Virtualization questions:
- Does virtualization have some effect on the HTML <img/> [loading complete](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/complete) attribute? 
- Is virtualization memoizing the component somehow? 
  - How is the old image even shown in a new component? (wouldn't the old image be discarded?). In both examples, browser cache was disabled. (though the problem remains with cache enabled too).
  - I've also tried setting react-virtual overscan to 0 (thinking it could be pre-fetching components), with no success.
  - react-virtual does not seem to be re-using components either, confirmed by a console.log() in each component's onMount useEffect lifecycle

I've tried many combinations of next versions, next/image (future and legacy), next/image props, and browsers. 
Interestingly, the same thing happens in react-virtuoso. (But not when removing virtualization, and simply mapping my data into jsx). 

Maybe I would find more answers in the nextjs discussions. But maybe someone here has insights on which parts of virtualization could be affecting the image component. Many thanks.

**Comment by michael-mcgreal** [comment]
Google mentions a good technique for virtualization is "[dom-recycling](https://developer.chrome.com/blog/infinite-scroller/#dom-recycling)". Maybe this is how react-virtual is reusing components, and showing the old card image until the new card image loads.  (Which somehow does not trigger the Next/Image component to show its blurPlaceholder value). 

Edit: asked in [NextJS discussions](https://github.com/vercel/next.js/discussions/41739)

**Reply by piecyk** [reply]
Hmm, is the `key` value the default one? if so, then pretty no way it's reusing anything as it's the index of element in array. Hard to say what is happening without minimal, reproducible example 🤔
